### DISCUSSION TITLE
deck.gl layer is not visible with Terrain

### DISCUSSION THREAD

**Discussion by karatemir** [maplibre/maplibre-gl-js#2245] [Q&A] [answered]
I am using MapLibre (version 2.2.0-pre.2) to render 3D terrain as described in [here](https://bertt.wordpress.com/2022/05/24/custom-maplibre-terrain-tiles/). I also have a Deck.gl [TripsLayer](https://deck.gl/docs/api-reference/geo-layers/trips-layer/) that I have added, basically a simple 3D trajectory describing some motion. The problem is I cannot see the TripLayer when terrain is on. However, if I toggle off the Terrain using the disable button, Deck.gl layer shows properly. 

Is this normal expected behavior?

**Comment by HarelM** [comment]
I'm guessing that the TripLayer is rendered inside the terrain or something.
Terrain obviously doesn't support everything...
Feel free to investigate how this should be made and share an example. :-)

**Comment by prozessor13** [comment]
As Harel said, i also assume that the trips layer is rendered below terrain (on elevation zero, sealevel). There may be two possible fixes:
* add terrain to deck.gl as well, and render the deck.gl layer above the maplibre map.
* implent render-to-texture for deck.gl layers and drape that textures onto maplibre terrain (i think is a lot of work, but may deck.gl has this functionallity already implemented, dont know)

**Comment by HarelM** [comment]
No example provided so there's no actual way to help here.
Converting this into a discussion.

**Comment by charlieforward9** [comment, marked as answer]
DeckGL offers the [Terrain Layer](https://deck.gl/docs/api-reference/geo-layers/terrain-layer), this may not have been available when this conversation was started, so just seeing here if anyone has found a method to this issue? I am attempting using the TerrainLayer with Mapbox's API's now but haven't been able to get it to work in globe mode. 

Since [DeckGL and Maplibre are now seamless](https://deck.gl/docs/whats-new#globe-view--maplibre) I was curious if there was an example available for this, or a strategy that I could consider to build an example for the community.

I would be happy to help with this at the beginning of 2026, if some guidance was offered.

**Reply by am2222** [reply]
@charlieforward9 did you manage to find any workaround for this?

**Reply by charlieforward9** [reply]
I got www.github.com/visgl/deck.gl/pull/10251 open to fix TerrainLayer on Globe - which may push this forward.

**Reply by am2222** [reply]
Thanks @charlieforward9 , just a quick question, does it mean the terrain layer should be added from the deckgl side or it will fixes the current maplibre+train and deckgl layers added on top?

**Reply by charlieforward9** [reply]
I'm focusing on deckgl-side terrain. If you have Maplibre terrain, you can try the TerrainExtension and #10252 has GlobeView fixes for this

**Reply by am2222** [reply]
Thanks @charlieforward9 ! I probably need to modify my current implementation.
just a side question, but is it possible to use deckgl terrain with interleaved mode of maplibre integration?

**Reply by charlieforward9** [reply]
I think there's an example of that in deck.gl

**Reply by am2222** [reply]
Thanks! I will look into that
